### DISCUSSION TITLE
Skip.env ANDROID_PACKAGE_NAME = painless.password

### DISCUSSION THREAD

**Discussion by davidakoontz** [skiptools/skip#203] [Q&A] [answered]
in the Skip.env (root folder) the Android package name has a dot in the name "painless.password".

I don't know where this comes from - or how it got set - but I've never used the dot notation for the App name.

I'm getting error:  The file "Resources" couldn't be saved in the folder "password" because a file with the same name already exist.
An error occurred while performing transpiration: The file "Resources" couldn't be saved in the folder "password" because a file with the same name already exists.

This is in an attempt to run the TESTS...

I can not find the folder "password"  but assume the path is in the derivedData folder and is because of the dot notation "painless.password"  All the other folders have abs paths.  This one does NOT!

Help,
David

**Comment by davidakoontz** [comment]
<img width="547" alt="Screenshot 2024-08-22 at 9 59 47 PM" src="https://github.com/user-attachments/assets/45f7186b-f2c9-4598-b8cd-1f0476a35f72">

Here the log file has a PainlessPassword/src/main/resources/painless/**password**/Resources/Module.xcassets/AppIcon.appiconset/  (*.png)

**Comment by davidakoontz** [comment]
<img width="1148" alt="Screenshot 2024-08-22 at 10 12 47 PM" src="https://github.com/user-attachments/assets/c8af4866-74c5-483a-8961-5540933933ec">

It looks like I've found the "painless" and "password" folders - up above the DerivedData folder ... in the <project>/Android folder.

**Comment by davidakoontz** [comment]
The only file in there is Main.kt  with the "package painless.password"  directive.

<img width="1153" alt="Screenshot 2024-08-22 at 10 17 32 PM" src="https://github.com/user-attachments/assets/56c9349e-05a7-4685-9d33-c54d2e61080b">

**Comment by marcprux** [comment, marked as answer]
Skip will derive Kotlin package names from a Swift module name by de-camel-casing the name and forming a valid Java package identifier. Thus, "PainlessPassword" converts to "painless.password". This is discussed in the docs on [platform customization](https://skip.tools/docs/platformcustomization/#kotlin-files). The module resources are embedded in your app as part of the owning package, which is why the Module.xcassets in your Resources/ folder will try to be copied and linked over to a destination folder of `src/main/resources/painless/password/Resources/Module.xcassets`.

That being said, I'm not sure where that error message is coming from. Did anything get renamed in the project recently that triggered this build/test error? You may want to try a clean build, and if that doesn't work, completely removing your DerivedData folder and re-starting Xcode, and that may clean things up.

**Reply by davidakoontz** [reply]
Thank you - really appreciate the effort that has gone into writing the Docs.  and pointing me to them!  I wish I had them all committed to memory/understanding... may in a year.

Anything get renamed... not that I can recall.  I will try the clean slate and restart.

**Reply by davidakoontz** [reply]
oh wait!  - renaming - yeah - that is me - my process involves constantly renaming things until they stick.

Now that I've walked my dog Malibu, I realize that I have a constant process of renaming everything... so my trying to fix things (after some renaming) is most likely making the complex build system unhappy.  Having two master's to please (Kotlin & Swift) is going to cause ME problems.  So now, I think I will resolve my problems - that are trending to more that ONE... if I start over with a Skip Init.  Use some better names,  drop in my source & test files a few at a time... and get back to a HAPPY building system.

**Reply by davidakoontz** [reply]
I had renamed the top project level folder... as well as many many other things...
and the App name "Painless Password" is going to cause endless issues - if I don't solve for the problem now... start over without the camel case App name - that gets Java-ified into ".../painless/password/..."

Yeah ... let me start again at  > skip init
